mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3584,17 +3625,18 @@ int main(int argc, char **argv) {
       llvm::errs() << "Host logic populated failed.\n";
       exit(1);
     }
-    if (applyBufferizationPipeline.getValue()) {
-      PassManager pm(module->getName(), PassManager::Nesting::Implicit);
+  }
 
-      rock::BufferizeOptions bufferizeOptions;
-      bufferizeOptions.disableRock = true;
-      rock::buildBufferizePipeline(pm, bufferizeOptions);
+  if (applyBufferizationPipeline.getValue()) {
+    PassManager pm(module->getName(), PassManager::Nesting::Implicit);
 
-      if (failed(pm.run(module))) {
-        llvm::errs() << "failed to apply rocm bufferize pipeline.\n";
-        exit(1);
-      }
+    rock::BufferizeOptions bufferizeOptions;
+    bufferizeOptions.disableRock = true;
+    rock::buildBufferizePipeline(pm, bufferizeOptions);
+
+    if (failed(pm.run(module))) {
```

Comment:
Why're we making this unconditional here?
I'm a little bit dubious about putting this in `rocmlir-gen`. I'd rather rip this bit of code out and stick it as `-host-pipeline bufferization` over in `rocmlir-driver`

Reply:
Previously, we were only using tosa in host harness.. now we are using in rock ops themselves.
I dont see good reason why it needs to be conditional given that when we generating kernel or host code -- we rather prefer them to be buffered -- either by construction and/or lowering.

Im fine either way -- just wanted to keep the contract rocmlir-gen to always produce bufferized IR as it is currently.

Reply:
Fair enough, reasonable point, let's not go aggressively refactoring things in this PR.

I'm thinking a future PR could be making `rocmlir-gen` output tensor code.